includes/REST_API/Products_Controller.php

```diff
@@ -0,0 +1,489 @@
+
+		if ( \is_string( $data ) && ! empty( $data ) ) {
+			/**
+			 * Decoded cached products data.
+			 *
+			 * @var array|null $products
+			 */
+			$products = json_decode( $data, true );
+
+			if ( $products ) {
+				return rest_ensure_response( $products );
+			}
+		}
+
+		$results = [];
+
+		/**
+		 * Shopify host.
+		 *
+		 * @var string $host
+		 */
+		$host = $this->settings->get_setting( Settings::SETTING_NAME_SHOPIFY_HOST );
+
+		/**
+		 * Shopify access token.
+		 *
+		 * @var string $access_token
+		 */
+		$access_token = $this->settings->get_setting( Settings::SETTING_NAME_SHOPIFY_ACCESS_TOKEN );
+
+		// TODO(#11154): Maybe move to a class constant.
+		$api_version = '2022-01';
+
+		// TODO(#11154): Proper sanitization.
+		$url = sprintf(
+			'https://%1$s/api/%2$s/graphql.json',
+			$host,
+			$api_version
+		);
+
+		$search_string = empty( $search_term ) ? '*' : '*' . $search_term . '*';
+
+		// TODO(#11154): Support different sortKeys.
+		$query = <<<QUERY
+{
+  products(first: 100, sortKey: CREATED_AT, query: "available_for_sale:true AND title:$search_string") {
+    edges {
+      node {
+        id
+        title
+        vendor
+        description
+        priceRange {
+          minVariantPrice {
+            amount
+            currencyCode
+          }
+        }
+        onlineStoreUrl
+        images(first: 10) {
+          edges {
+            node {
+              url
+              altText
+            }
+          }
+        }
+      }
+    }
+  }
+}
+QUERY;
+
+		$response = wp_remote_post(
+			$url,
+			[
+				'headers' => [
+					'Content-Type'                      => 'application/graphql',
+					'X-Shopify-Storefront-Access-Token' => $access_token,
+				],
+				'body'    => $query,
+			]
+		);
+
+		if ( WP_Http::OK !== wp_remote_retrieve_response_code( $response ) ) {
+			return new \WP_Error( 'rest_unknown', __( 'Error fetching products', 'web-stories' ), [ 'status' => 404 ] );
+		}
+
+		/**
+		 * Shopify GraphQL API response.
+		 *
+		 * @var ShopifyGraphQLResponse $result
+		 */
+		$result = json_decode( wp_remote_retrieve_body( $response ), true );
+
+		// TODO: Error handling.
+		if ( isset( $result['errors'] ) ) {
+			return new \WP_Error( 'rest_unknown', __( 'Error fetching products', 'web-stories' ), [ 'status' => 404 ] );
+		}
+
+		foreach ( $result['data']['products']['edges'] as $edge ) {
+			$product = $edge['node'];
+
+			$images = [];
+
+			foreach ( $product['images']['edges'] as $image_edge ) {
+				$image    = $image_edge['node'];
+				$images[] = [
+					'url' => $image['url'],
+					'alt' => $image['altText'],
+				];
+			}
+
+			$results[] = [
+				'productId'            => $product['id'],
+				'productTitle'         => $product['title'],
+				'productBrand'         => $product['vendor'],
+				// TODO: Maybe eventually provide full price range.
+				// See https://github.com/ampproject/amphtml/issues/37957.
+				'productPrice'         => $product['priceRange']['minVariantPrice']['amount'],
```

Comment:
How does this work? What if you have a product that has variants from $1 to $1000. Isn't this prices from $1? Is there anyway to show there are variants and prices may different. 

Worth noting that some countries is a little matter if you do not show the correct price next to the item.

Reply:
Product ranges are not supported yet by the Web Stories format itself in, which is why this TODO links to https://github.com/ampproject/amphtml/issues/37957. Once the format supports it, this will need updating.

There's nothing we can do about it right now.